seatunnel-transforms-v2/src/main/java/org/apache/seatunnel/transform/filter/FilterFieldTransform.java

```diff
@@ -36,21 +36,39 @@
 
 import java.util.ArrayList;
 import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
 import java.util.stream.Collectors;
+import java.util.stream.Stream;
 
 @Slf4j
 public class FilterFieldTransform extends AbstractCatalogSupportTransform {
     public static final String PLUGIN_NAME = "Filter";
-    private int[] inputValueIndex;
-    private final List<String> fields;
+
+    private int[] inputValueIndexList;
+
+    private final List<String> includeFields;
+    private final List<String> excludeFields;
 
     public FilterFieldTransform(
             @NonNull ReadonlyConfig config, @NonNull CatalogTable catalogTable) {
         super(catalogTable);
         SeaTunnelRowType seaTunnelRowType = catalogTable.getTableSchema().toPhysicalRowDataType();
-        fields = config.get(FilterFieldTransformConfig.KEY_FIELDS);
+        includeFields = config.get(FilterFieldTransformConfig.INCLUDE_FIELDS);
+        excludeFields = config.get(FilterFieldTransformConfig.EXCLUDE_FIELDS);
+        // only one should be set
+        if (Objects.nonNull(includeFields) && Objects.nonNull(excludeFields)) {
```

Comment:
use `ConfigValidator.of(config).validate()` replace this code

```
      if (Objects.nonNull(includeFields) && Objects.nonNull(excludeFields)) {
          throw TransformCommonError.bothIncludeAndExcludeFieldsError(getPluginName());
      }
      if (Objects.isNull(includeFields) && Objects.isNull(excludeFields)) {
          throw TransformCommonError.noIncludeNorExcludeFieldsError(getPluginName());
      }
```

Reply:
fixed, PTAL again @EricJoy2048